### PR TITLE
cmake: Update pmdk version to 1.7 

### PR DESCRIPTION
**Pull request by CongMinYin**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by CongMinYin** [comment]
@tchaikov Currently I just used PMDK repository in the code. Should I continue to use ceph/nvml repository?  In fact the nvml is renamed to pmdk. Should I create a PMDK reposity in ceph?

File: cmake/modules/Buildpmem.cmake [hunk 1]
```diff
@@ -12,8 +12,8 @@ function(build_pmem)
   set(PMDK_LIB "${PMDK_SRC}/${PMDK_LIB_DIR}")
 
   ExternalProject_Add(nvml_ext
-      GIT_REPOSITORY "https://github.com/ceph/nvml.git"
-      GIT_TAG "dd622819dd4ee97d3920f913c70be"
+      GIT_REPOSITORY "https://github.com/pmem/pmdk.git"
```
Comment (tchaikov) [inline cmake/modules/Buildpmem.cmake]: s/pmem/ceph/ see https://github.com/ceph/pmdk

Comment (tchaikov) [reply cmake/modules/Buildpmem.cmake]: and please note the rationale in the commit message, for instance

1. nvml is renamed to pmdk, and
2. we need to bump to the latest stable release: 1.7 ..

Comment (lixiaoy1) [reply cmake/modules/Buildpmem.cmake]: > s/pmem/ceph/ see https://github.com/ceph/pmdk

@tchaikov Hi Kefu, is this what you want to update PMDK repository in Ceph? https://github.com/ceph/pmdk/pull/2

Comment (tchaikov) [reply cmake/modules/Buildpmem.cmake]: as discussed offline. i just sync'ed ceph/pmdk. so we just need to point `GIT_REPOSITORY` to it and update `GIT_TAG` to `1.8`.

Comment (dillaman) [reply cmake/modules/Buildpmem.cmake]: +1 to use named tag `stable-1.8` if possible instead of commit hash

File: cmake/modules/Buildpmem.cmake [hunk 2]
```diff
@@ -12,8 +12,8 @@ function(build_pmem)
...
+      GIT_TAG "bc5e309485df61c452d08367e4b13ba9dfed5071"
       SOURCE_DIR ${CMAKE_BINARY_DIR}/src/nvml
```
Comment (dillaman) [inline cmake/modules/Buildpmem.cmake]: Should we be updating all the `nvml` directory references to `pmdk` as well?

File: cmake/modules/Buildpmem.cmake (hunk 1 above)
Comment (tchaikov) [reply cmake/modules/Buildpmem.cmake]: @CongMinYin could you take this opportunity to s/nvml_ext/pmdk_ext/ as well?

**Comment by CongMinYin** [comment]
@tchaikov 
① nvml is renamed pmdk，changed the nvml repository to pmdk repository ,
② update pmdk GIT_TAG to 1.7
③ renamed the directory "nvml" to "pmdk"

File: cmake/modules/Buildpmem.cmake (hunk 1 above)
Comment (CongMinYin) [reply cmake/modules/Buildpmem.cmake]: > @CongMinYin could you take this opportunity to s/nvml_ext/pmdk_ext/ as well?

@tchaikov Yes, I am testing it now.

**Comment by tchaikov** [comment]
@CongMinYin i mean, the external project is now named "nvml_ext", could you rename it to "pmem_ext" in this change?

**Comment by CongMinYin** [comment]
> @CongMinYin i mean, the external project is now named "nvml_ext", could you rename it to "pmem_ext" in this change?
@tchaikov  Yes, I mean I renamed it to "pmdk_ext" and do test now. Should I renamed it to "pmem_ext"?

**Comment by tchaikov** [comment]
> > @CongMinYin i mean, the external project is now named "nvml_ext", could you rename it to "pmem_ext" in this change?
> 
> @tchaikov Yes, I mean I renamed it to "pmdk_ext" and do test now. Should I renamed it to "pmem_ext"?

oh, i see. and yeah, i agree with you. `pmdk_ext` is better.

**Comment by CongMinYin** [comment]
> > > @CongMinYin i mean, the external project is now named "nvml_ext", could you rename it to "pmem_ext" in this change?
> > 
> > 
> > @tchaikov Yes, I mean I renamed it to "pmdk_ext" and do test now. Should I renamed it to "pmem_ext"?
> 
> oh, i see. and yeah, i agree with you. `pmdk_ext` is better.

@tchaikov I have renamed "nvml_ext " to "pmdk_ext" and tested.

**Pull request merged**